maitake/src/wait/cell.rs

```diff
@@ -189,6 +229,32 @@ impl fmt::Debug for WaitCell {
     }
 }
 
+// === impl Wait ===
+
+impl Future for Wait<'_> {
+    type Output = Result<(), Error>;
```

Comment:
this never returns `Error::Busy`...I think it should probably be changed to return the `wait::Closed` error type, rather than `Error`.

if we're going to add a variant of this future that behaves differently when the cell is busy, I think it would make sense for that future to have a different error type.